samples/sample_camera/include/pipeline_camera.h

```diff
@@ -0,0 +1,192 @@
+
+This sample was distributed or derived from the Intel's Media Samples package.
+The original version of this sample may be obtained from https://software.intel.com/en-us/intel-media-server-studio
+or https://software.intel.com/en-us/media-client-solutions-support.
+\**********************************************************************************/
+
+#ifndef __PIPELINE_DECODE_H__
+#define __PIPELINE_DECODE_H__
+
+#include "sample_defs.h"
+
+#if D3D_SURFACES_SUPPORT
+#pragma warning(disable : 4201)
+#include <d3d9.h>
+#include <dxva2api.h>
+#endif
+
+#include <vector>
+#include "hw_device.h"
+#include "camera_render.h"
+#include <memory>
+
+#include "sample_camera_utils.h"
+
+#include "sample_defs.h"
+#include "sample_utils.h"
+#include "base_allocator.h"
+
+#include "mfxplugin.h"
+#include "mfxplugin++.h"
+#include "mfxvideo.h"
+#include "mfxvideo++.h"
+
+#include "vm/thread_defs.h"
+
+// The only supported padding size is 8
+#define CAMERA_PADDING_SIZE 8
+
+#define align(value) ((0x10) * ( (value) / (0x10) + (((value) % (0x10)) ? 1 : 0)))
+#define align_32(value) ((0x20) * ( (value) / (0x20) + (((value) % (0x20)) ? 1 : 0)))
+#define align_64(value) ((0x40) * ( (value) / (0x40) + (((value) % (0x40)) ? 1 : 0)))
+
+class CCameraPipeline
+{
+public:
+
+    CCameraPipeline();
+    virtual ~CCameraPipeline();
+
+    virtual mfxStatus Init(sInputParams *pParams);
+    virtual mfxStatus Run();
+    virtual void Close();
+    virtual mfxStatus Reset(sInputParams *pParams);
+
+    void SetExtBuffersFlag()       { m_bIsExtBuffers = true; }
+    void SetRenderingFlag()        { m_bIsRender = true; }
+    void SetOutputfileFlag(bool b) { m_bOutput = b; }
+    virtual void PrintInfo();
+
+    mfxU32 GetNumberProcessedFrames() {return m_nFrameIndex;}
+
+    mfxStatus PrepareInputSurfaces();
+
+protected:
+
+    bool isBayerFormat(mfxU32 type);
+
+    mfxCam3DLutEntry*       m_3dlut_17;
+    mfxCam3DLutEntry*       m_3dlut_33;
+    mfxCam3DLutEntry*       m_3dlut_65;
+    CBmpWriter*             m_pBmpWriter;
+    CVideoReader*           m_pFileReader;
+    CRawVideoWriter*        m_pRawFileWriter;
+    mfxU32                  m_nInputFileIndex;
+    mfxU32                  m_nFrameIndex; // index of processed frame
+    mfxU32                  m_nFrameLimit; // limit number of frames to proceed
+
+    MFXVideoSession     m_mfxSession;
+    MFXVideoVPP*        m_pmfxVPP;
+    mfxVideoParam       m_mfxVideoParams;
+    std::auto_ptr<MFXVideoUSER>  m_pUserModule;
+    std::auto_ptr<MFXPlugin> m_pCamera_plugin;
```

Comment:
better to change all ```auto_ptr``` to ```unique_ptr```

Reply:
auto_ptr is deprecated API. This will give warning if anyone will try to bring the code up on Linux. I think this needs to be fixed.

Reply:
This code is for Windows only. I agree that this might be changed to unique_ptr, but let's do it in a separate pull request as you told us before. The purpose of this patch is to merge currently existing (and working) windows code without much change.

Reply:
Filed #1317